### PR TITLE
[release-1.25] OCPBUGS-25908: Cherry-pick changes from containers/storage/pull#1641

### PR DESCRIPTION
**Pull request by kwilczynski**
#### What type of PR is this?

/kind bug
/assign kwilczynski

#### What this PR does / why we need it:

Manually cherry-pick changes from https://github.com/containers/storage/pull/1641 as these changes contain a fixe that needs to be backported to CRI-O release 1.25, part of OpenShift 4.12 release.

Related:

- https://github.com/containers/storage/pull/1641
- https://github.com/cri-o/cri-o/pull/7171
- https://github.com/cri-o/cri-o/pull/7536

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -17,15 +17,15 @@ require (
...
 	github.com/containers/conmon v2.0.20+incompatible
 	github.com/containers/conmon-rs v0.4.0
-	github.com/containers/image/v5 v5.25.0
+	github.com/containers/image/v5 v5.24.0
 	github.com/containers/kubensmnt v1.2.0
 	github.com/containers/ocicrypt v1.1.7
-	github.com/containers/podman/v4 v4.5.2-0.20230706090613-813f1b53bf39
-	github.com/containers/storage v1.46.1
+	github.com/containers/podman/v4 v4.4.1
+	github.com/containers/storage v1.45.7-0.20240109024041-ee156648626e
```
[the remark this answers is not in the document]
Comment (kwilczynski) [reply go.mod:28]: @haircommander OK!

[the remark this answers is not in the document]
Comment (kwilczynski) [reply go.mod]: Done.

**Review by mrunalp — APPROVED** [review]
/lgtm

**Pull request merged**